src/components/TodoList.tsx

```diff
@@ -0,0 +1,270 @@
+import React, { useEffect, useRef, useState } from 'react';
+import {
+  CSSTransition,
+  TransitionGroup,
+} from 'react-transition-group';
+import {
+  changeTodoTitle,
+  deleteTodo,
+  getTodos,
+  postTodo,
+  toggleCompleteTodo,
+} from '../api/todos';
+import { Todo } from '../types/Todo';
+import { Header } from './TodoList/Header';
+import { Footer } from './TodoList/Footer';
+import { TodoItem } from './TodoList/TodoItem';
+import { FilterType } from '../types/FilterType';
+import { ErrorNotification } from './TodoList/ErrorNotification';
+
+type Props = {
+  userId: number;
+};
+
+export const TodoList: React.FC<Props> = ({ userId }) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [filteredTodo, setFilteredTodo] = useState<FilterType>(FilterType.ALL);
+  const [processingTodoIds, setProcessingTodoIds] = useState<number[]>([]);
+  const [focusToHeaderInput, setFocusToHeaderInput] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  useEffect(() => {
+    getTodos(userId)
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+        setTimeout(() => {
+          setErrorMessage('');
+        }, 3000);
+      })
+      .finally(() => {
+        setProcessingTodoIds([]);
+      });
+  }, [userId]);
+
+  let updatedTodos = [...todos];
+
+  switch (filteredTodo) {
+    case FilterType.ALL:
+      updatedTodos = todos.filter(todo => todo);
+      break;
+    case FilterType.ACTIVE:
+      updatedTodos = todos.filter(todo => !todo.completed);
+      break;
+    case FilterType.COMPLETED:
+      updatedTodos = todos.filter(todo => todo.completed);
+      break;
+    default: updatedTodos = todos;
+  }
```

Comment:
let's use `useState` and `useEffect` for this

Reply:
rewrote this part with useEffect, thanks)